storage/imxrt-flash/flashdrv.c

```diff
@@ -26,15 +27,109 @@
 #define QSPI_FREQ_133MHZ 0xc0000008
 
 
+#define QUAD_FAST_READ_SEQ_ID      0
+#define READ_STATUS_REG_SEQ_ID     1
+#define WRITE_ENABLE_SEQ_ID        3
+#define SECTOR_ERASE_SEQ_ID        5
+#define BLOCK_ERASE_SEQ_ID         8
+#define PAGE_PROGRAM_SEQ_ID        9
+#define CHIP_ERASE_SEQ_ID          11
+#define READ_JEDEC_ID_SEQ_ID       12
+
+
+static int flash_setWEL(flash_context_t *context, uint32_t dstAddr)
+{
+	int err = EOK;
+	flexspi_xfer_t xfer;
+
+	xfer.baseAddress = dstAddr;
+	xfer.operation = kFlexSpiOperation_Command;
+	xfer.seqId = WRITE_ENABLE_SEQ_ID;
+	xfer.seqNum = 1;
+	xfer.isParallelModeEnable = 0;
+
+	if ((err = flexspi_norFlashExecuteSeq(context->instance, &xfer)) != 0)
+		return -err;
+
+	return err;
+}
+
+
+static int flash_writeBytes(flash_context_t *context, uint32_t dstAddr, uint32_t *src, uint32_t size)
+{
+	int err = EOK;
+	flexspi_xfer_t xfer;
+
+	xfer.txSize = size;
+	xfer.txBuffer = src;
+	xfer.baseAddress = dstAddr;
+	xfer.operation = kFlexSpiOperation_Write;
+	xfer.seqId = PAGE_PROGRAM_SEQ_ID;
+	xfer.seqNum = 1;
+
+	if ((err = flexspi_norFlashExecuteSeq(context->instance, &xfer)) != 0)
+		return -err;
+
+	return err;
+}
+
+
+static int flash_waitBusBusy(flash_context_t *context)
+{
+	int err = EOK;
+	uint32_t buff;
+	uint8_t retrans;
+	flexspi_xfer_t xfer;
+	const uint8_t MAX_RETRANS = 8;
+
+	retrans = 0;
+	xfer.rxSize = 4;
+	xfer.rxBuffer = &buff;
+	xfer.baseAddress = context->instance;
+	xfer.operation = kFlexSpiOperation_Read;
+	xfer.seqId = READ_STATUS_REG_SEQ_ID;
+	xfer.seqNum = 1;
+	xfer.isParallelModeEnable = 0;
+
+	do
+	{
+		err = flexspi_norFlashExecuteSeq(context->instance, &xfer);
+		usleep(100);
```

Comment:
imxrt have no high precision clock. 100 will be truncated to 0, so no sleep at all